### PR TITLE
Rewrite the <launchable> tag when using rename-desktop-file

### PR DESCRIPTION
**Pull request by hughsie**
This copies FlatpakXml from the flatpak project.

Fixes https://github.com/flatpak/flatpak-builder/issues/155

### CODE REVIEW

File: src/builder-manifest.c [hunk 1]
```diff
@@ -2420,35 +2420,54 @@ builder_manifest_cleanup (BuilderManifest *self,
...
-                  g_string_append (new_contents, replace_dst);
-                  to_replace = match + strlen (replace_src);
+                  g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_FAILED, "no <component> node");
+                  return FALSE;
+                }
+              n_id = flatpak_xml_find (n_root, "id", NULL);
+              if (n_id)
+                {
+                  n_text = n_id->first_child;
+                  if (n_text && g_strcmp0 (n_text->text, self->id) != 0)
```
Comment (alexlarsson) [inline src/builder-manifest.c]: I don't think this is quite right. There can be multiple components. This happens for instance in libreoffice which has components for each sub-app (draw, calc, writer, etc). I think we should only rewrite the ones that match the original desktop file, no?

Comment (hughsie) [reply src/builder-manifest.c]: so more `if (n_text && g_strcmp0 (n_text->text, self->rename_desktop_file) == 0)`?

Comment (hughsie) [reply src/builder-manifest.c]: Fixed, and commit message improved to reflect reality.

**Comment by alexlarsson** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**